drivers/si1145/si1145_saul.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2016 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_si1145
+ * @{
+ *
+ * @file
+ * @brief       SAUL adaption for SI1145 device
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "saul.h"
+#include "si1145.h"
+#include "xtimer.h"
+
+static uint16_t uv, ir, visible, distance;
+
+static int check_and_read_uv(void *dev, phydat_t *res)
+{
+    si1145_t *d = (si1145_t *)dev;
+
+    si1145_read_uv(d, &uv);
+    res->val[0] = uv;
+    res->unit = UNIT_NONE;
+    res->scale = 0;
+    return 1;
+}
+
+static int check_and_read_ir(void *dev, phydat_t *res)
+{
+    si1145_t *d = (si1145_t *)dev;
+
+    si1145_read_ir(d, &ir);
+
+    res->unit = UNIT_LUX;
+    res->scale = 0;
+    return 1;
+}
+
+static int check_and_read_visible(void *dev, phydat_t *res)
+{
+    si1145_t *d = (si1145_t *)dev;
+
+    si1145_read_ir(d, &visible);
+
+    res->unit = UNIT_LUX;
+    res->scale = 0;
+    return 1;
+}
+
+static int check_and_read_distance(void *dev, phydat_t *res)
+{
+    si1145_t *d = (si1145_t *)dev;
+
+    si1145_read_distance(d, &distance);
+
+    res->unit = UNIT_M;
+    res->scale = 0;
+    return 1;
+}
+
+
+static int read_uv(void *dev, phydat_t *res)
```

Comment:
What is the advantage of invoking another method with the exact same arguments. Compiler will optimize it out, but it increases LoC.